src/ast/irbuilderbpf.cpp

```diff
@@ -2479,12 +2479,22 @@ CallInst *IRBuilderBPF::CreateSkbOutput(Value *skb,
   return call;
 }
 
-Value *IRBuilderBPF::CreatKFuncArg(Value *ctx,
-                                   SizedType &type,
-                                   std::string &name)
+Value *IRBuilderBPF::CreateKFuncArg(Value *ctx,
+                                    SizedType &type,
+                                    std::string &name)
 {
   assert(type.IsIntTy() || type.IsPtrTy());
+
+#if LLVM_VERSION_MAJOR <= 14
+  // Older versions of LLVM check that the type matches the GEP. Newer versions
+  // do not use this cast because we need to specifically preserve the pattern
+  // `GEP(preserveStaticOffsets(ctx))`, which doesn't exist on older versions.
```

Comment:
So the pointer cast would drop the `preserve_static_offsets` attribute off `ctx`, right?

What if we used `llvm.bpf.getelementptr.and.load` directly? Wouldn't that simplify the code as we wouldn't need these ifdefs? I'm just curious, not sure if that's even possible.